src/agent/src/main.rs

```diff
@@ -394,6 +416,99 @@ async fn start_sandbox(
     Ok(())
 }
 
+// Start-up attestation-agent, CDH and api-server-rest if they are packaged in the rootfs
+fn init_attestation_components(logger: &Logger, _config: &AgentConfig) -> Result<()> {
+    // The Attestation Agent will run for the duration of the guest.
+    launch_process(
+        logger,
+        AA_PATH,
+        &vec!["--attestation_sock", AA_ATTESTATION_URI],
+        AA_ATTESTATION_SOCKET,
+        DEFAULT_LAUNCH_PROCESS_TIMEOUT,
+    )
+    .map_err(|e| anyhow!("launch_process {} failed: {:?}", AA_PATH, e))?;
+
+    let config_path = OCICRYPT_CONFIG_PATH;
+
+    // The image will need to be encrypted using a keyprovider
+    // that has the same name (at least according to the config).
+    let ocicrypt_config = serde_json::json!({
+        "key-providers": {
+            "attestation-agent":{
+                "ttrpc":CDH_SOCKET
+            }
+        }
+    });
+
+    fs::write(config_path, ocicrypt_config.to_string().as_bytes())?;
+
+    env::set_var("OCICRYPT_KEYPROVIDER_CONFIG", config_path);
+
+    if let Err(e) = launch_process(
+        logger,
+        CDH_PATH,
+        &vec![],
+        CDH_SOCKET,
+        DEFAULT_LAUNCH_PROCESS_TIMEOUT,
+    ) {
+        error!(logger, "launch_process {} failed: {:?}", CDH_PATH, e);
+    } else if !_config.rest_api.is_empty() {
```

Comment:
I think I mentioned this already, but I wonder if we should change the default option to be enabling the ASR with attestation only (rather than not starting it at all). This way people wouldn't be required to set the `rest_api` param in most cases.

Reply:
Hey Tobin, just to double check I understand, is the proposed improve to set the default value of `rest_api` in the agent config.rs to be `attestation` rather than empty?

Reply:
Wait I typed exactly the opposite of what I should have. The default would be `resource` not `attestation`. I think updating the parameter name isn't a bad idea.

Reply:
Thanks - I was slightly confused about why we are using `resource` in the test, but my brain is tired, so I'm easily persuaded! I'll fix that up now.